armi/physics/neutronics/fissionProductModel/fissionProductModel.py

```diff
@@ -152,44 +148,61 @@ def _fissionProductBlockType(self):
 
     def interactBOL(self):
         interfaces.Interface.interactBOL(self)
-        self.setAllBlockLFPs()
+        if self._explicitFissionProducts:
+            self.setAllComponentFissionProducts()
+        else:
+            self.setAllBlockLFPs()
+
+    def setAllComponentFissionProducts(self):
+        """
+        Initialize all blueprint nuclides for each ``DEPLETABLE`` component in the core.
+
+        Notes
+        -----
+        This should be called when explicit fission product modeling is enabled to
+        ensure that fission products are initialized on the depletable components within
+        the reactor data model.
+
+        .. note::
+
+            When explicit fission products are enabled and the user has not already included
+            all fission products in the blueprints (in ``nuclideFlags``), the ``fpModelLibrary`` setting is used
+            to autofill all the nuclides in a given library into the ``blueprints.allNuclidesInProblem``
+            list. All nuclides that were not manually initialized by the user are added to
+            the ``DEPLETABLE`` components throughout every block in the core. The ``DEPLETABLE``
+            flag is based on the user adding this explicitly in the blueprints, or is based on
+            the user setting a nuclide to ``burn: true`` in the blueprint ``nuclideFlags``.
+
+        See Also
+        --------
+        armi.reactor.blueprints.isotopicOptions.autoUpdateNuclideFlags
+        armi.reactor.blueprints.isotopicOptions.getAllNuclideBasesByLibrary
+        """
+        for b in self.r.core.getBlocks(includeAll=True):
+            b.setLumpedFissionProducts(None)
+            for c in b.getComponents(Flags.DEPLETABLE):
+                ndens = c.getNumberDensities()
+                updatedNDens = {}
```

Comment:
`updatedNDens` should start as `ndens`, not as an empty dict, right?
```suggestion
                updatedNDens = c.getNumberDensities()
```